src/AspNetCoreMulti/Example/GraphQL/Queries.cs

```diff
@@ -0,0 +1,38 @@
+using GraphQL.Types;
+using System.Collections.Generic;
+
+namespace Example.GraphQL
+{
+    public class Queries
+    {
+        public class DogQuery : ObjectGraphType<object>
+        {
+            public DogQuery(IEnumerable<IDogOperation> dogOperations)
+            {
+                foreach(var dogOperation in dogOperations)
+                {
+                    var fields = dogOperation.RegisterFields();
+                    foreach(var field in fields)
+                    {
+                        AddField((FieldType)field);
+                    }
+                }
+            }
+        }
+
+        public class CatQuery : ObjectGraphType<object>
+        {
+            public CatQuery(IEnumerable<ICatOperation> catOperations)
+            {
+                foreach(var catOperation in catOperations)
+                {
+                    var fields = catOperation.RegisterFields();
+                    foreach (var field in fields)
+                    {
+                        AddField((FieldType)field);
```

Comment:
Maybe it's better to use `AutoSchema` here.

Reply:
Can you give me some more details here? Do you have any doc handy for `AutoSchema`?  I'm thinking it is some way to set things up automatically.

One reason I am doing the way I am right now, is to be non-opinionated on the setup. I give control to the implementor of `IQuery` to give me the fields as they see fit. They can subsequently use reflection or any other controlled mechanism to set things up and finally return the fields which I would then attach to the root query. This is especially useful when the application is large, and consumers need several ways to opt in/out of things. 

Again, I might be talking about something different but will wait for information on `AutoSchema`.

Reply:
I'm not really sure how the use of `AutoSchema` fits into your example.  `AutoSchema` is useful when creating a type-first schema, where the schema is automatically constructed from CLR types.  Your sample does not use those features.  Moreover, you are attempting to use methods from multiple classes to assemble a large root graph type.  An auto-generated root query type would be contrary to that goal.

Reply:
You can see a demonstration of a type-first schema here:

https://github.com/graphql-dotnet/graphql-dotnet/tree/master/src/GraphQL.StarWars.TypeFirst

The entire schema can be constructed via:

```cs
services.AddGraphQL(b => b
    .AddSystemTextJson()
    .AddAutoSchema<StarWarsQuery>(c => c.WithMutation<StarWarsMutation>()));
```

You will notice there are no 'graph types' but rather simple CLR classes

Reply:
In my own graphs, I use a slight variation of this to allow for DI injection of scoped services, which is not possible with the sample seen there.  However, the code exists in the tests here:

https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.Tests/Bugs/Issue2932_DemoDIGraphType.cs

I also published it as a NuGet package here:

https://www.nuget.org/packages/Shane32.GraphQL.DI

Reply:
@EmmanuelPonnudurai I'm fine with example as is without `AutoSchema`.